scipy/_lib/_array_api.py

```diff
@@ -223,12 +226,40 @@ def xp_copy(x: Array, *, xp: ModuleType | None = None) -> Array:
     return _asarray(x, copy=True, xp=xp)
 
 
+_default_xp: ContextVar[ModuleType | None] = ContextVar("_default_xp", default=None)
```

Comment:
From https://docs.python.org/3/library/contextvars.html: 

> Context managers that have state should use Context Variables instead of [threading.local()](https://docs.python.org/3/library/threading.html#threading.local) to prevent their state from bleeding to other code unexpectedly, when used in concurrent code.

and it's used in a context manager below. TIL something.

Reply:
I would expect any failure in contextvars caused by nogil to be treated as a bug in the cpython interpreter

Reply:
Thanks, agreed. I'm going to take a closer look on this PR in the coming days.